crates/augurs-prophet/src/features.rs

```diff
@@ -13,71 +15,119 @@ pub enum FeatureMode {
+        Self { start, end }
     }
 
-    /// Set the lower window for the holiday.
+    /// Create a new holiday encompassing midnight on the day
+    /// of the given timestamp to midnight on the following day,
+    /// in UTC.
     ///
-    /// The lower window is the number of days before the holiday
-    /// that it is observed. For example, if the holiday is on
-    /// 2023-01-01 and the lower window is 1, then the holiday will
-    /// _also_ be observed on 2022-12-31.
-    pub fn with_lower_window(mut self, lower_window: Vec<u32>) -> Result<Self, Error> {
-        if self.ds.len() != lower_window.len() {
-            return Err(Error::MismatchedLengths {
-                a_name: "ds".to_string(),
-                a: self.ds.len(),
-                b_name: "lower_window".to_string(),
-                b: lower_window.len(),
-            });
-        }
-        self.lower_window = Some(lower_window);
-        Ok(self)
+    /// This is a convenience method to reproduce the behaviour
+    /// of the Python and R Prophet implementations, which require
+    /// all holidays to be day-long events.
+    ///
+    /// Note that this will _not_ handle daylight saving time
+    /// transitions correctly. To handle this correctly, use
+    /// [`HolidayOccurrence::new`] with the correct start and
+    /// end times, e.g. by calculating them using [`chrono`].
+    ///
+    /// [`chrono`]: https://docs.rs/chrono/latest/chrono
+    pub fn for_day(day: TimestampSeconds) -> Self {
+        Self::for_day_in_tz(day, 0)
     }
 
-    /// Set the upper window for the holiday.
+    /// Create a new holiday encompassing midnight on the day
+    /// of the given timestamp to midnight on the following day,
+    /// in a timezone represented by the `utc_offset`.
+    ///
+    /// This is a convenience method to reproduce the behaviour
+    /// of the Python and R Prophet implementations, which require
+    /// all holidays to be day-long events.
     ///
-    /// The upper window is the number of days after the holiday
-    /// that it is observed. For example, if the holiday is on
-    /// 2023-01-01 and the upper window is 1, then the holiday will
-    /// _also_ be observed on 2023-01-02.
-    pub fn with_upper_window(mut self, upper_window: Vec<u32>) -> Result<Self, Error> {
-        if self.ds.len() != upper_window.len() {
-            return Err(Error::MismatchedLengths {
-                a_name: "ds".to_string(),
-                a: self.ds.len(),
-                b_name: "upper_window".to_string(),
-                b: upper_window.len(),
-            });
+    /// Note that this will _not_ handle daylight saving time
+    /// transitions correctly. To handle this correctly, use
+    /// [`HolidayOccurrence::new`] with the correct start and
+    /// end times, e.g. by calculating them using [`chrono`].
+    ///
+    /// [`chrono`]: https://docs.rs/chrono/latest/chrono
+    pub fn for_day_in_tz(day: TimestampSeconds, utc_offset: i32) -> Self {
```

Comment:
`utc_offset` is very unintuitive here, reading the code it's a seconds offset for the TZ I'm in. So UTC+1, this param should be 3600?

Reply:
Ok I saw your comment on the class that goes into this. It's weird though!